### ISSUE TITLE
boto dependency on ec2 instance creation (ssl proxy)

### ISSUE THREAD

**Issue by tbugfinder** [ansible-community/molecule-ec2#37]
Hi,

I'm having issues provisioning an EC2 instance due to its dependency on boto instead of boto3 while running behind ssl proxy.
At least that's my understanding:

Error:
```
vations", "  File \"/usr/local/lib/python3.6/site-packages/boto/vpc/__init__.py\", line 1152, in get_all_subnets", "    return self.get_list('DescribeSubnets', params, [('item', Subnet)])", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 1170, in get_list", "    response = self.make_request(action, params, path, verb)", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 1116, in make_request", "    return self._mexe(http_request)", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 913, in _mexe", "    self.is_secure)", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 705, in get_http_connection", "    return self.new_http_connection(host, port, is_secure)", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 747, in new_http_connection", "    connection = self.proxy_ssl(host, is_secure and 443 or 80)", "  File \"/usr/local/lib/python3.6/site-packages/boto/connection.py\", line 796, in proxy_ssl", "    sock.sendall(\"CONNECT %s HTTP/1.0\\r\\n\" % host)", "TypeError: a bytes-like object is required, not 'str'"]}
```

As per my understanding this kind of errors are getting to known like:
https://github.com/boto/boto/pull/3699

**Comment by tbugfinder** [comment]
Workaround seems to be using the delegated driver with community ec2 module.
```
molecule.yml:

driver:
  name: delegated
```

with:
```
    - name: Create molecule instance(s)
      ec2:
      community.aws.ec2_instance:
```

**Comment by tbugfinder** [comment]
After some reading, I'd also understand that molecule-ec2 should move from amazon.aws.ec2 (depends on boto, get bugfixes but no features) to community.aws.ec2_instance.

https://docs.ansible.com/ansible/latest/collections/amazon/aws/ec2_module.html#ansible-collections-amazon-aws-ec2-module

https://github.com/ansible-collections/community.aws/blob/main/docs/community.aws.ec2_instance_module.rst

**Comment by jgoldschrafe** [comment]
You fixed this in #38, right?
